### ISSUE TITLE
Autocommand errors for TextChanged/TextChangedI

### ISSUE THREAD

**Issue by frnsys** [L3MON4D3/LuaSnip#1034]
Hi, thanks for your plugin. I recently started having this issue with LuaSnip and can't figure out why (I'm on the latest version).

```
Error detected while processing TextChangedI Autocommands for "*":                                                                                                    
Error executing lua callback: .../.local/share/nvim/lazy/LuaSnip/lua/luasnip/util/log.lua:105: attempt to index local 'msg' (a nil value)                             
stack traceback:                                                                                                                                                      
        .../.local/share/nvim/lazy/LuaSnip/lua/luasnip/util/log.lua:105: in function 'warn'                                                                           
        ...seng/.local/share/nvim/lazy/LuaSnip/lua/luasnip/init.lua:490: in function <...seng/.local/share/nvim/lazy/LuaSnip/lua/luasnip/init.lua:472>
```

I'm using it with `nvim-cmp` and it's configured like so:

```
      require('luasnip').setup({
        history = true,
        update_events = "TextChanged,TextChangedI",
      })
```

This is what my snippets file looks like:

```
local ls = require('luasnip')

local s = ls.s
local i = ls.insert_node
local f = ls.function_node
local rep = require("luasnip.extras").rep
local fmt = require("luasnip.extras.fmt").fmt

-- i(1) means insert at the first position
-- i(0) means insert at the last position
ls.add_snippets("all", {
  -- Insert current datetime
  s("now", f(function()
    return os.date("%m.%d.%Y %H:%M")
  end))
})

--- HTML
local html_snippets = {
  s("<a>", fmt("<a href=\"{}\">{}</a>", { i(1), i(2) }))
}
ls.add_snippets("html", html_snippets);
ls.add_snippets("typescriptreact", html_snippets);

--- Markdown
ls.add_snippets("markdown", {
  -- To do item
  s("o", fmt("- [ ] {}", { i(1) }))
})

--- Python
ls.add_snippets("python", {
  s("l", fmt("print(\"{}\");", { i(1) })),
  s("d", fmt("import ipdb; ipdb.set_trace()", {}))
})

--- C#
ls.add_snippets("cs", {
  -- Logging
  s("l", fmt("Debug.Log(\"{}\");", { i(1) })),
  s("lv", fmt("Debug.Log(\"{}: \" + {});", { i(1), i(2) })),

  -- Function/method definitions
  s("void", fmt([[
  {} {}({}) {{
      {}
  }}
  ]], { i(1, "void"), i(2, "Function"), i(3), i(4) })),
  s("pub", fmt([[
  public {} {}({}) {{
      {}
  }}
  ]], { i(1, "void"), i(2, "Function"), i(3), i(4) })),

  -- Loops
  s("foreach", fmt([[
  foreach (var {} in {}) {{
      {}
  }}
  ]], { i(1), i(2), i(3) })),

  -- Conditionals
  s("nc", fmt([[
  if ({} == null) return;
    ]], { i(1) })),

  s("if", fmt([[
    if ({}) {{
        {}
    }}
    ]], { i(1), i(2) })),

  s("elif", fmt([[
  else if ({}) {{
      {}
  }}
  ]], { i(1), i(2) })),

  s("else", fmt([[
else {{
    {}
}}
]], { i(1) }))
})

--- Rust
ls.add_snippets("rust", {
  s("l", fmt("println!(\"{}\");", { i(1) })),
  s("ld", fmt("println!(\"{{}}\", {});", { i(1) })),
  s("lv", fmt("println!(\"{} {{:?}}\", {});", { i(1), rep(1) })),

  s("readfile", fmt([[
  let {} = fs::read_to_string({})
      .unwrap_or_else(|_| panic!("Couldn't read file: {{:?}}", {}));
  ]], { i(1, "contents"), i(2, "&path"), rep(2) })),

  s("writefile", fmt([[
    fs::write({}, {})
      .unwrap_or_else(|_| panic!("Couldn't write file: {{:?}}", {}));
  ]], { i(2, "&path"), i(1, "data"), rep(2) })),

  s("tests", fmt([[
  #[cfg(test)]
  mod tests {{
      use super::*;

      {}
  }}
  ]], { i(0) })),

  s("test", fmt([[
  #[test]
  fn test_{}() {{
      {}
  }}
  ]], { i(1), i(0) }))
})
```

I don't know if this is related, but I also started having an issue where, after using a snippet, I'd start to get a persistent multiple-cursor like behavior in the area around the snippet. Lines would be duplicated and I'd have to restart nvim for insert behavior to return to normal.

**Comment by L3MON4D3** [comment]
Ahhh yup, I missed that, thank you!

> I don't know if this is related, but I also started having an issue where, after using a snippet, I'd start to get a persistent multiple-cursor like behavior in the area around the snippet. Lines would be duplicated and I'd have to restart nvim for insert behavior to return to normal.

Could you record that the next time it happens? I'd expected that specifically issues like this should be fixed with the last few commits :((

**Comment by L3MON4D3** [comment]
(BTW, fixed in 3cfa0c7f11ba8c02c5637bf4821647b74235d6c3) :sweat_smile:

**Comment by frnsys** [comment]
Wow, thank you so much for the quick fix! I will definitely catch the other bug the next time it occurs and reopen an issue for it.

Thanks again :)
